src/pose.rs

```diff
@@ -0,0 +1,249 @@
+use crate::config::{Color, PoseListenerConfig};
+use crate::transformation::ros_pose_to_isometry;
+use nalgebra::geometry::{Isometry3, Point3};
+use std::option::Option;
+use std::sync::{Arc, RwLock};
+use tui::style;
+use tui::widgets::canvas::Line;
+
+use rosrust;
+
+fn pose_to_arrow(pose: &Isometry3<f64>, length: f64, color: &Color) -> Vec<Line> {
+    let mut lines: Vec<Line> = Vec::new();
+    let tui_color = style::Color::Rgb(color.r, color.g, color.b);
+    let pt1 = pose.transform_point(&Point3::new(0.0, 0.0, 0.0));
+    let pt2 = pose.transform_point(&Point3::new(length, 0.0, 0.0));
+    lines.push(Line {
+        x1: pt1.x,
+        y1: pt1.y,
+        x2: pt2.x,
+        y2: pt2.y,
+        color: tui_color,
+    });
+    let pt_left = pose.transform_point(&Point3::new(length / 4.0 * 2.0, length / 4.0, 0.0));
+    lines.push(Line {
+        x1: pt2.x,
+        y1: pt2.y,
+        x2: pt_left.x,
+        y2: pt_left.y,
+        color: tui_color,
+    });
+    let pt_right = pose.transform_point(&Point3::new(length / 4.0 * 2.0, -length / 4.0, 0.0));
+    lines.push(Line {
+        x1: pt2.x,
+        y1: pt2.y,
+        x2: pt_right.x,
+        y2: pt_right.y,
+        color: tui_color,
+    });
+    lines
+}
+
+fn pose_to_axes(pose: &Isometry3<f64>, length: f64) -> Vec<Line> {
```

Comment:
did you test it with odd orientations? looks like it works, but i would like to see it ;) this could be helpful if we want to show the tf tree

Reply:
It should work with no problem. I didn't try to rotate it in very weird ways, but the z axis seems to be properly represented ;)